### PR TITLE
Storage module and Solr configuration

### PR DESCRIPTION
**Pull request by houssam-mahlous**
- Save resources that are read from kafka topics to Solr documents
- Save mentions to Solr documents instead of Postgresql
- Change mention ID type from `Long` to `String`

### CODE REVIEW

File: storage/src/main/java/com/brandwatch/minibcr/storage/services/MentionStoreService.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.brandwatch.minibcr.common.domain.Mention;
+import com.brandwatch.minibcr.common.repository.mention.MentionRepository;
+
+@Service
+public class MentionStoreService {
```
Comment (kocevaIv) [inline storage/src/main/java/com/brandwatch/minibcr/storage/services/MentionStoreService.java:11]: Where do you use this service?

File: common/src/main/java/com/brandwatch/minibcr/common/domain/Mention.java [hunk 2]
```diff
@@ -1,38 +1,124 @@
...
+            this.createdAt = createdAt;
+            return this;
+        }
+
+        public Mention build() {
+            return new Mention(mentionId, queryId, text, fromUser, createdAt);
+        }
     }
 }
+
```
Comment (kocevaIv) [inline common/src/main/java/com/brandwatch/minibcr/common/domain/Mention.java]: Unnecessary new line at the end.

**Review by kocevaIv — APPROVED** [review]
Great work! Everything looks good apart from few comments i left.

File: storage/src/main/java/com/brandwatch/minibcr/storage/services/MentionStoreService.java (hunk 1 above)
Comment (houssam-mahlous) [reply storage/src/main/java/com/brandwatch/minibcr/storage/services/MentionStoreService.java:11]: Nowhere at the moment but I am guessing when a Mentions Consumer is created, the service will be used to save whatever mentions that are passed through kafka.

**Pull request merged**